components/style/properties/helpers.mako.rs

```diff
@@ -294,27 +312,34 @@
                         % if property.logical:
                             let wm = context.style.writing_mode;
                         % endif
-                        <% maybe_wm = ", wm" if property.logical else "" %>
+                        <%
+                            maybe_wm = ", wm" if property.logical else ""
+                            maybe_cacheable = ", cacheable" if property.has_uncacheable_values else ""
+                        %>
                         match *value {
                             DeclaredValue::Value(ref specified_value) => {
                                 % if property.ident in SYSTEM_FONT_LONGHANDS and product == "gecko":
                                     if let Some(sf) = specified_value.get_system() {
                                         longhands::system_font::resolve_system_font(sf, context);
                                     }
                                 % endif
-                                let computed = specified_value.to_computed_value(context);
-                                % if property.ident == "font_size":
-                                    longhands::font_size::cascade_specified_font_size(context,
-                                                                                      specified_value,
-                                                                                      computed,
-                                                                                      inherited_style.get_font());
+                                % if property.is_vector and product == "gecko":
+                                    let mut s = context.mutate_style().take_${data.current_style_struct.name_lower}();
```

Comment:
(Well, actually I guess they can use `self`, so we'd be good there too).

In general I don't love the complexity this patch introduces overall, but since I don't have a better suggestion right now, other than the other comments...